config/base.yml

```
@@ -29,6 +29,13 @@ cache:
     - $HOME/.composer/cache/files
 
 before_script:
+  # Dynamically work out COMPOSER_ROOT_VERSION:
+  # Numeric branch - 4 / 4.8 => 4.x-dev / 4.8.x-dev
+  # Tag - 1.2.3 / 1.2.3-beta1 => 1.2.3 / 1.2.3-beta1
+  # Non-numeric branch - master / some-branch => dev-master / dev-some-branch
+  # $TRAVIS_BRANCH documentation - https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+  - if [[ "$TRAVIS_BRANCH" =~ ^[1-9]$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev" elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}" else export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}" fi
```

This one-liner threw an error for me when I tried running it locally. I had to add a `;` at the end of each export statement to get in working

```suggestion
  - if [[ "$TRAVIS_BRANCH" =~ ^[1-9]$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"; elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}"; else export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}"; fi
```

I think there's a way you can break a statement over multiple line in YML. I would rather we try that for clarity. https://www.baeldung.com/yaml-multi-line

---

opps, have updated

I'd prefer to keep on a single so that it remains consistent with the rest of the file, where there's already plenty of long single lines.  Also I don't know how well the log in travis, which adds little javascript arrows to toggle expand/collapse output, would respond.